app/src/utils/useDatasetStore.ts

```diff
@@ -172,3 +172,16 @@ export const DatasetProvider = DatasetStoreContext.Provider;
 export default function useDatasetStore(): DatasetStore {
   return useContext(DatasetStoreContext);
 }
+
+/**
+ * @param callback Callback that is executed exactly once, once metadata is
+ *     loaded.
+ */
+export function useMetadataLoaded(callback: (metadata: MetadataMap) => void) {
```

Comment:
I think useMetadataLoaded is somewhat a confusing function name... I'm not sure I have a better suggestion though. Maybe useMetadataLoadedCallback or afterMetadataLoaded? I think "use" prefix is kind of confusing but I get using it to match the hooks pattern. Feel free to leave as is just some rambling thoughts :)

Reply:
Updated to useOnMetadataLoaded, hopefully that clarifies things a bit